sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -621,6 +731,10 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                         // 2. break out of the loop to allow requesting another session from the service
                         if (IsSessionProcessor && _sessionId == null)
                         {
+                            await OnSessionClosingAsync(
+                                (ServiceBusSessionReceiver)receiver,
+                                cancellationToken)
+                                .ConfigureAwait(false);
```

Comment:
Lets say we create a receiver-1 and If we are done with receiving messages from 1 session, then create a new receiver and start receiving messages from second session and then we create receiver-3, receiver-4 and so on. How do we dispose all the receivers?

Reply:
The receiver gets disposed on [line 775](https://github.com/JoshLove-msft/azure-sdk-for-net/blob/processor-settle/sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs#L775)

Reply:
Ah..finally block.